### PR TITLE
(feat) Boxcraft integrated with Boostlet

### PR DESCRIPTION
**Pull request by shrutivarade**
1. This PR consists of all the code related to Testing and Boxcraft.
2. Boxcraft is currently working only for OSD and CACTAS.

### CODE REVIEW

File: src/frameworks/xtk.js [hunk 1]
```diff
@@ -56,34 +51,37 @@ export class Xtk extends Framework {
...
+    // end of flip
 
     let masked_image = Util.harden_mask(image.data, new_mask);
 
-    let masked_image_as_imagedata = new ImageData(masked_image, newcanvas.width, newcanvas.height);
-
-    ctx.putImageData(masked_image_as_imagedata, 0, 0); // rgba data, no flip
-
-    originalcanvas.parentNode.replaceChild(newcanvas, originalcanvas);
+    this.set_image(masked_image); // rgba data, no flip
```
Comment (gaiborjosue) [inline src/frameworks/xtk.js:84]: Did you test this .set_image as well?

File: src/frameworks/xtk.js [hunk 2]
```diff
@@ -15,13 +15,8 @@ export class Xtk extends Framework {
     let image = ctx.getImageData(0, 0, canvas.width, canvas.height);
     let rgba_image = Util.rgba_to_grayscale(image.data);
 
-    if(from_canvas) {
-      return { data: image.data, width: image.width, height: image.height };
-    }
-    else {
-      return { data: rgba_image, width: image.width, height: image.height };
+    return { data: rgba_image, width: image.width, height: image.height };
```
Comment (gaiborjosue) [inline src/frameworks/xtk.js:18]: Hello @shrutivarade, why did you get rid of the different return features? If I remember correctly, this allowed it to work correctly, just out of curiosity :)

**Review by gaiborjosue — COMMENTED** [review]
Hello @shrutivarade I have tested the version for xtk that you suggest, however it doesn't work. It seems it is the older version we had posted on the main branch, maybe I would suggest syncing it with the current xtk.js file to avoid any issues.

Also, I tried to to segment anything with the working version of xtk (current one on the main branch) but it doesn't work. I guess it is still under development right?

**Comment by haehn** [comment]
> Hello @shrutivarade, I have tested the version for xtk you suggested; however, it doesn't work. It seems it is the older version we had posted on the main branch; maybe I would suggest syncing it with the current xtk.js file to avoid any issues :).
> 
> Also, I tried segment anything with the working version of xtk (the current one on the main branch), but it doesn't work. I guess it is still under development, right?

Hi @shrutivarade whats the status on using the new xtk code?

**Comment by shrutivarade** [comment]
@haehn I reverted the code from the working version, and now xtk is working with clicks.

I also completed the test setup for Niivue(CACTAS) and OSD with boxcraft, and it is working well.
Now I am working on getting xtk work with boxcraft

**Pull request closed without merging**